### PR TITLE
ncm-network: add ipv6 support to the nmstate backend

### PR DESCRIPTION
**Pull request by guillaume-philippon**
Add ipv6 support to the nmstate backend supporting:
- Interfaces with a single IPv6 address
- Interfaces with a single IPv4 and a single IPv6 address
- Default route only

### CODE REVIEW

**Comment by jrha** [comment]
Thanks! I'll get this tested at RAL.

**Comment by guillaume-philippon** [comment]
Last version should allow ipv4 and ipv6 on same interface (was not the case before) but not allow aliases.

**Comment by aka7** [comment]
> Last version should allow ipv4 and ipv6 on same interface (was not the case before) but not allow aliases.

@guillaume-philippon  I wonder if you can pull in the alias bits I got in my pr into this? to support for both ipv4 n ipv6? so can all go into your pr if its not too much trouble? or I can wait until yours is merged to add Alias support.
not sure why the tests are failing on both though, unrelated I think.

**Comment by guillaume-philippon** [comment]
I think it would be easiest to have only one PR. Especially if some code refactoring are needded.

**Comment by jrha** [comment]
This functions well on test hosts at RAL, with the limitations I've added to the description text.

The tests are failing because no tests seem to be executed by `nmstate_ipv6.t`, which is odd because it looks reasonable to me.

**Comment by aka7** [comment]
@guillaume-philippon can you rebase and publish again,  @jrha James has merged my PR https://github.com/quattor/configuration-modules-core/pull/1688 
so this  will need a rebase.

**Comment by jrha** [comment]
I'll have a go at rebasing this.

File: ncm-network/src/main/perl/nmstate.pm
Comment (guillaume-philippon) [reply ncm-network/src/main/perl/nmstate.pm:485]: push didn't work well. It was some testing for ipv6 aliases

**Comment by jrha** [comment]
Rebased and re-tested at RAL, looks good. I'll see if I can work out why the tests are unhappy.

File: ncm-network/src/main/perl/nmstate.pm
Comment (jrha) [reply ncm-network/src/main/perl/nmstate.pm:485]: Ok, I'll see if I can solve this too.

**Comment by jrha** [comment]
I've sorted out the issue with the tests and finished the implementation of secondary IPv6 addresses.

The tests are now only failing due to the pre-existing issue with `ncm-sysctl`.

**Comment by jrha** [comment]
I'll open another PR to implement additional route support for IPv6.

**Review by jrha — APPROVED** [review]

**Pull request merged**
